CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+cmake_minimum_required(VERSION 3.15)
```

Comment:
I'm thinking about wanting this CMake file included or not. On one hand, it's good work and definitely nice to have, but on the other hand it might soon get out of date without me dedicating extra time to maintain it too. It is not too complex (the Sharpmake version is a bit more complicated, but could also be used to output a makefile I believe).

Reply:
The other thing is that we really only need the NetImguiServer application to have multi-platform compiling support. The samples are just a simple demonstration that can easily be compiled without a build system configured, and the NetImgui client code is usually included in an existing project, and is the one that was meant to be truly multi-platform.